web/src/actions/clusters.ts

```diff
@@ -0,0 +1,52 @@
+'use server';
+
+import { KubeConfig } from '@/lib/k8s-client';
+
+export interface ClusterInfo {
+  name: string;
+  endpoint: string;
+  source: string; // Which environment variable it came from
+}
+
+export async function getClusters(): Promise<ClusterInfo[]> {
+  const clusters: ClusterInfo[] = [];
+  
+  // List of environment variables to check for kubeconfig
+  const envVars = ['KUBECONFIG_PRIMARY', 'KUBECONFIG_FOO', 'KUBECONFIG_BAR'];
```

Comment:
We should find all envs that start with KUBECONFIG_ and get the suffix for the name. eg this should be a dynamic list

Reply:
Create the kube config should be done like this in k8s-client it should done at startup to avoid creating the apiClient over and over.